### DISCUSSION TITLE
Can't transform T4 templates because FileNotFoundException occurs

### DISCUSSION THREAD

**Discussion by lsouzavms** [serenity-is/Serenity#6435] [Help]
Hi, guys

My project has a problem where every time I run the command "Transform all T4 templates", it doesn't finish executing because the following error appears:
Running the transformation: System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the specified file.
File name: 'Newtonsoft.Json, Version=11.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed'.
The file specified where the error occurs is ClientTypes.tt

From what I understand, the folder that includes the Serenity.CodeGenerator has a Newtonsoft.Json dll in that same version, so I assume he might not be able to find it.
And worse, I sent this same project to a colleague to know if the problem was in the project or in my machine, and he saw that it gave the same error, but it affected another project he works on.
I imagine the dll used in the project and the one used by CodeGenerator might be different as this project has always used Newtonsoft.Json 12.0.3 while Serenity.CodeGenerator uses 11.0.2. I just don't understand how this project gave such a problem, because I was messing with it without problems and without changing anything in the packages, and suddenly it gave this.
Thanks to anyone who can help.

I'm using Visual Studio 2017, all packages of Serenity are in version 3.9.6, my OS is Windows 10 Pro

**Comment by lsouzavms** [comment]
Update: I tried to get my project without any modifications from a repository, just to see if the problem is something I may have messed with, but with no luck. Something happened with my machine after that, which is even more strange, because another project I have in the same pc runs T4 transformations all fine.

**Comment by VictorTomaili** [comment]
We didn't use the t4 transformation anymore. The old version of code generation is obsolete. This means when the used libraries like newtonsoft upgraded in your project, it may give exception depending breaking changes etc. 
The better option is upgrade the project to the .net 5 or .net 6 version. You can find the upgrade documentation in this link [https://serenity.is/docs](https://serenity.is/docs).

**Reply by lsouzavms** [reply]
It's weird, because I never needed to update the packages used in this project. All the versions I have of the packages already came like this from the repository I downloaded it from. I'm kind of insecure about having to update the project because it was working normally before, so I don't know if it has something to do with the version.

**Comment by stixoffire** [comment]
The issue is sometimes caused by an error in your typescript such as a missing bracket , semicolon comma or just a typo etc..
Check your error listings and warnings . If that is not the case verify that the referenced library does indeed exist in the project and is set for inclusion upon compile. 
Also given you are using VS 2017 - the possibility exists something got messed up in the project references - so remove it (close vs 2017 - reopen) and add it back in again .

2017 has some wacky behaviors. I have seen issues like this before and it can be a pain in the but.
